exercises/ml-basics/ex_rnw/sol_ml_proj.Rnw

```diff
@@ -1 +1,124 @@
+  feature coefficients. This does not change our model formulation, but we have 
+  to keep in mind that it implicitly entails adding an element 1 at the first 
+  position of each feature vector, constituting the familiar column of ones 
+  in the design matrix $\Xmat$.
+  
+  \item The parameter space is included in the definition of the hypothesis 
+  space and in this case given by $\Theta = \R^3$.
+  
+  \item Loss function for the $i$-th observation: $\Lxyit = \left( \yi - 
+  \thetab^T \xi \right)^2$.
+  
+  \item The first thing to note is that both MLE and ERM are 
+  \textbf{optimization problems}, and both should lead us to the same optimum. 
+  Their opposite signs are not a problem: maximizing the likelihood is 
+  equivalent to minimizing the negative likelihood. 
+  Also, both are defined pointwise.
+  The last thing to fix is therefore the product introduced by the independence 
+  assumption in the joint likelihood of all observations (recall that we use 
+  a \textit{summed} loss in ERM), for which the logarithm is a natural remedy.
+  We can thus simply use the \textbf{negative log-likelihood} as our loss 
+  function (and indeed, many known loss functions can be shown to correspond to 
+  certain model likelihoods).
+  
+  Let's put these reflections to practice:
+  
+  \begin{flalign*}
+    \Lxyit &= - \log \LL(\thetab | \xi) \\
+    &= - \ll(\thetab | \xi) \\
+    &= - \log \left(\frac{1}{\sqrt{2 \pi 
+    \sigma^2}} \exp \left(- \frac{1}{2 \sigma^2} \left( \yi - \thetab^T \xi  
+    \right)^2  \right) \right) \\
+    &= - \left(\log \left( \frac{1}{\sqrt{2 \pi \sigma^2}} \right) + 
+    \log \left( \exp \left(- \frac{1}{2 \sigma^2} \left( \yi - \thetab^T \xi  
+    \right)^2  \right) \right) \right)  \\
+    &= - \left(- \frac{1}{2}\log(2 \pi \sigma^2) - \frac{1}{2 \sigma^2} \left( 
+    \yi - \thetab^T \xi \right)^2 \right) \\
+    &= \frac{1}{2}\log(2 \pi \sigma^2) + \frac{1}{2 \sigma^2} \left( 
+    \yi - \thetab^T \xi \right)^2
+  \end{flalign*}
+  
+  \begin{flalign*}
+    \risket &= \sumin - \ll(\thetab | \xi) \\
+    &= \sumin \Lxyit \\
+    &= \sumin \frac{1}{2} \log(2 \pi \sigma^2) + \frac{1}{2 \sigma^2} 
+    \left(\yi - \thetab^T \xi \right)^2 \\
+    &= \frac{n}{2}\log(2 \pi \sigma^2) +  \frac{1}{2 \sigma^2} \sumin
+    \left(\yi - \thetab^T \xi \right)^2 \\
+    &\propto \sumin \left(\yi - \thetab^T \xi \right)^2
+  \end{flalign*}  
+  
+  As we are only interested in the feature coefficients here, we neglect all 
+  irrelevant terms that do not depend on $\thetab$ as they have no effect on
+  the solution.
+  This is what the proportional sign $\propto$, often used in 
+  contexts of optimization and Bayesian statistics, means: we keep 
+  only expressions impacted by our parameter of interest because they suffice 
+  to yield the intended results or show some property of interest.
+  
+  From this we can easily see the correspondence between MLE and ERM.
```

Comment:
am ende: "The L2 loss is proportional to the negative log-likelihood and hence, the argmax of the likelihood (using the assumption of normally distributed errors) and the argmin of the risk (using L2 loss) are equal."